tools/golang-json-rpc-tests/main.go

```diff
@@ -0,0 +1,412 @@
+    }
+}
+
+func testFeeHistory(client *ethclient.Client, blockCount uint64, newestBlock *big.Int, rewardPercentiles []float64) {
+    ctx := context.Background()
+
+    feeHistory, err := client.FeeHistory(ctx, blockCount, newestBlock, rewardPercentiles)
+    if err != nil {
+        log.Fatalf("Failed to get fee history: %v", err)
+    }
+
+    if feeHistory == nil {
+        log.Fatalf("Fee history is nil")
+    }
+
+    fmt.Printf("Fee History:\n")
+    fmt.Printf("Oldest Block: %s\n", feeHistory.OldestBlock.String())
+    fmt.Printf("Gas Used Ratio: %v\n", feeHistory.GasUsedRatio)
+    fmt.Printf("Reward: %v\n", feeHistory.Reward)
+}
+
+func waitForTransaction(client *ethclient.Client, tx *types.Transaction) *types.Receipt {
+    ctx := context.Background()
+    receipt, err := bind.WaitMined(ctx, client, tx)
+    if err != nil {
+        log.Fatalf("Failed to wait for transaction to be mined: %v", err)
+    }
+    if receipt == nil {
+        log.Fatalf("Receipt is nil")
+    }
+    return receipt
+}
+
+func testGetBlockTransactionCountByHash(client *ethclient.Client, blockHash common.Hash) {
+    txCount, err := client.TransactionCount(context.Background(), blockHash)
+    if err != nil {
+        log.Fatalf("Failed to get transaction count by block hash: %v", err)
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block hash %s: %d\n", blockHash.Hex(), txCount)
+}
+
+func testGetBlockTransactionCountByNumber(client *ethclient.Client) {
+    txCount, err := client.PendingTransactionCount(context.Background())
+    if err != nil {
+        log.Fatalf("Failed to get transaction count by block number: %v", err)
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block number: %d\n", txCount)
+}
+
+func testCodeAt(client *ethclient.Client, address common.Address) {
+    code, err := client.CodeAt(context.Background(), address, nil)
+    if err != nil {
+        log.Fatalf("Failed to get code at address: %v", err)
+    }
+    if code == nil {
+        log.Fatalf("Code is nil")
+    }
+    fmt.Printf("Code at address %s: %s\n", address.Hex(), hex.EncodeToString(code)) // Not a SmartContract so empty string expected
+}
+
+func testGetLogs(client *ethclient.Client, address common.Address, topics []common.Hash) {
+    query := ethereum.FilterQuery{
+        Addresses: []common.Address{address},
+        Topics:    [][]common.Hash{topics},
+    }
+
+    logs, err := client.FilterLogs(context.Background(), query)
+    if err != nil {
+        log.Fatalf("Failed to get logs: %v", err)
+    }
+
+    fmt.Printf("Logs for address %s:\n", address.Hex())
+    if len(logs) == 0 {
+        fmt.Printf("No logs found for address %s\n", address.Hex())
+    } else {
+        for _, vLog := range logs { // Not a SmartContract so empty result expected
```

Comment:
The SC deployed using eth_sendRawTransaction can be useful here

Reply:
Yes, this sample SC emits a log, so the response is no longer empty here:

https://github.com/hashgraph/hedera-json-rpc-relay/pull/2586#discussion_r1650705127